toolchain/lower/file_context.h

```diff
@@ -165,6 +184,69 @@ class FileContext {
 
   auto BuildVtable(const SemIR::Class& class_info) -> llvm::GlobalVariable*;
 
+  // Track which specifics were lowered for each generic. These are added one
+  // by one while lowering their definitions.
+  auto AddLoweredSpecificForGeneric(SemIR::GenericId generic_id,
+                                    SemIR::SpecificId specific_id) {
+    lowered_specifics_[generic_id.index].push_back(specific_id);
+  }
+
+  // Initialize a return a SpecificFunctionFingerprint* instance for specific.
+  // The internal of the fingerprint are populated during and after lowering
+  // the function body of that specific.
+  auto InitializeFingerprintForSpecific(SemIR::SpecificId specific_id)
+      -> SpecificFunctionFingerprint* {
+    if (!specific_id.has_value()) {
+      return nullptr;
+    }
+    return &lowered_specific_fingerprint_[specific_id.index];
+  }
+
+  // Entry point for coalescing equivalent specifics. Two function definitions,
```

Comment:
Just a note, but it might be worth thinking about how these "Entry point" and "while coalescing" comments could be removed by shifting to a helper class. Understood as a todo, just want to flag since the "while coalescing" comments in particular mean that the comments on what a function is doing is somewhat buried behind that condition.